cmd/detectExecuteScan.go

```diff
@@ -0,0 +1,70 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+func detectExecuteScan(myDetectExecuteScanOptions detectExecuteScanOptions) error {
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
```

Comment:
Basically not against this, it is up to you. Redirecting the output from the layer below like this results in some more the less ugly output, e.g.

```
time="2019-12-05T08:02:45Z" level=error msg="An error occured. Stdout from underlying process: >>\nAPI_URL: https://xx.xx.xx.xx:30030\nThe authenticity of host 'xx.xx.xx.xx' is not validated!\nUSERNAME: ****\nAuthenticating...\nORG: PM\nSPACE: ****\nAPI endpoint:   https://xx.xx.xx.xx:30030 (API version: 1)\nUser:           ****\nOrg:            PM\nSpace:          ****\n\nNAME:\n   bg-deploy - Deploy a multi-target app using blue-green deployment\n\nUSAGE:\n   Deploy a multi-target app using blue-green deployment\n   xs bg-deploy MTA [-e EXT_DESCRIPTOR[,...]] [-u URL] [-t TIMEOUT] \n   [-v VERSION_RULE] [-f] [--no-start] [--use-namespaces] [--no-namespaces-for-services] \n   [--delete-services] [--delete-service-keys] [--delete-service-brokers] [--keep-files] \n   [--no-restart-subscribed-apps] [--do-not-fail-on-missing-permissions] [--abort-on-error] [--git-uri URI] \n   [--git-ref NAME] [--git-skip-ssl-validation] [--skip-ownership-validation] [--no-confirm] \n   \n   Interact with an active blue-green deploy operation\n   xs bg-deploy [-i OPERATION_ID] [-a ACTION] \n\nARGUMENTS:   \n   MTA       Multi-target app archive or directory path. e.g. 'myApp.mtar','myApp/' or '.' \n\nOPTIONS:                                  \n   -e <EXT_DESCRIPTOR[,...]>              Extension descriptors\n   -u <URL>                               Deploy service URL\n   -t <TIMEOUT>                           Start timeout in seconds\n   -v,--version-rule <VERSION_RULE>       Version rule (HIGHER, SAME_HIGHER, ALL)\n   -i <OPERATION_ID>                      Active deploy operation id\n   -a <ACTION>                            Action to perform on active deploy operation (abort, retry, monitor, resume)\n   -f                                     Force deploy without confirmation for aborting conflicting processes\n   --no-start                             Do not start apps\n   --use-namespaces                       Use namespaces in app and service names\n   --no-namespaces-for-services           Do not use namespaces in service names\n   --delete-services                      Recreate changed services / delete discontinued services\n   --delete-service-keys                  Delete existing service keys and apply the new ones\n   --delete-service-brokers               Delete discontinued service brokers\n   --keep-files                           Keep files used for deployment\n   --no-restart-subscribed-apps           Do not restart subscribed apps, updated during the deployment\n   --do-not-fail-on-missing-permissions   Do not fail on missing permissions for admin operations\n   --abort-on-error                       Auto-abort the process on any errors\n   --git-uri <URI>                        Git repository url or repository name in xsa embeded git service\n   --git-ref <NAME>                       Git repository branch, tag or ref name\n   --git-skip-ssl-validation              Skip git https ssl verification\n   --skip-ownership-validation            Skip the ownership validation that prevents the modification of entities managed by other multi-target apps\n   --no-confirm                           Do not require confirmation for deleting the previously deployed MTA apps\n\n\nLogging out...\nOK\n\n<<. Stderr from underlying process: >>Error while parsing command 'bg-deploy'\nUnknown option: does-not-exist\n\n<<" library= stepName=xsDeploy
```
Basically everything is provided as one log line with '\n' inbetween. Other approach: Using a scanner we could separate the log lines having a more readable log. Disadvantage of that approach: we have individual log lines which does not have a relationship in the log (might be desirable in case we forward the log to some logging frameworks). Yet another approach: We can simply forward to stderr without using the logging framework here (maybe the tool underneath already performed some log formatting.

But not requiring any change, just mentioning other appraoches in order to have it "on the table".

Reply:
Actually from what the tests show is that every log line ends up as a dedicated line in the logging.
From the current perspective this is what I targeted so far.

Reply:
Thats cool. I saw it the other way.